### PR TITLE
Support autodoc_type_aliases configuration

### PR DESCRIPTION
**Pull request by tikuma-lsuhsc**
This PR addresses Issues #284.

Currently, `autodoc-typehints` does not honor the [`autodoc_type_aliases` configuration of `sphinx.ext.autodoc`](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_type_aliases). For example,

```python
from numpy.typing import ArrayLike

def func(x: ArrayLike):...
```

renders

```rst
**func(x: _SupportsArray[dtype[Any]] |
_NestedSequence[_SupportsArray[dtype[Any]]] | bool | int | float |
complex | str | bytes | _NestedSequence[bool | int | float | complex |
str | bytes])
```

even if `autodoc_type_aliases = {'ArrayLike': 'ArrayLike'}` line in `conf.py`.

I modified `autodoc_typehints` to pass `app.config.autodoc_type_aliases` to `sphinx_signature()` calls to match `sphinx.ext.autodoc` and print the alias instead of the annotation as discovered.

`test_integration_autodoc_type_aliases.py` is a simple pytest script to check the conservation of `ArrayLike` annotation in a function argument.

### CODE REVIEW

**Review by gaborbernat — APPROVED** [review]

**Comment by tikuma-lsuhsc** [comment]
@gaborbernat - Could you help me sort out the tox env type check errors?

I'm getting

```
tests/test_integration_autodoc_type_aliases.py:22: error: "T" has no attribute "EXPECTED"  [attr-defined]
tests/test_integration_autodoc_type_aliases.py:23: error: "T" has no attribute "OPTIONS"  [attr-defined]
tests/test_integration_autodoc_type_aliases.py:31: error: "T" has no attribute "WARNING"  [attr-defined]
tests/test_integration_autodoc_type_aliases.py:56: error: Missing return statement  [empty-body]
```

I got the last one, but not clear about the others. I created the test file by copying then stripping away existing tests of the `test_integration.py`. Thanks

**Comment by gaborbernat** [comment]
Nothing jumps out for me from the top of my head. 🤔

**Comment by tikuma-lsuhsc** [comment]
OK. I'll look into it when I get home tonight.

**Comment by tikuma-lsuhsc** [comment]
Resolved the issue (I think).  The test file `test_integration.py` was excluded from the `mypy` type check via `pyproject.toml`. If you try to merge again, it should work now. Thanks.

File: tests/test_integration_autodoc_type_aliases.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    :param x: foo
+    :return: something
+    :rtype: bytes
+    """
+
+
+AUTO_FUNCTION = ".. autofunction:: mod.{}"
+
+LT_PY310 = sys.version_info < (3, 10)
```
Comment (gaborbernat) [inline tests/test_integration_autodoc_type_aliases.py:68]: Inline these two to where they are used.

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: tests/test_integration_autodoc_type_aliases.py (hunk 1 above)
Comment (tikuma-lsuhsc) [reply tests/test_integration_autodoc_type_aliases.py]: done

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**